includes/widgets/common.php

```diff
@@ -364,6 +409,247 @@ protected function register_controls() {
 
 		$this->end_controls_section();
 
+		$this->start_controls_section(
+			'_section_masking',
+			[
+				'label' => __( 'Mask', 'elementor' ),
+				'tab' => Controls_Manager::TAB_ADVANCED,
+			]
+		);
+
+		$this->add_control(
+			'_mask_switch',
+			[
+				'label' => __( 'Mask', 'elementor' ),
+				'type' => Controls_Manager::SWITCHER,
+				'label_on' => __( 'On', 'elementor' ),
+				'label_off' => __( 'Off', 'elementor' ),
+				'default' => '',
+			]
+		);
+
+		$options = array_merge(
+			$this->get_shapes(),
+			[ 'custom' => __( 'Custom', 'elementor' ) ]
+		);
+
+		$this->add_control( '_mask_shape', [
+			'label' => __( 'Shape', 'elementor' ),
+			'type' => Controls_Manager::SELECT,
+			'options' => $options,
+			'default' => $this->get_shape_url( 'circle' ),
+			'selectors' => $this->get_mask_selectors( '-webkit-mask-image: url( {{VALUE}} );' ),
+			'condition' => [
+				'_mask_switch!' => '',
+			],
+		] );
+
+		$this->add_control(
+			'_mask_image',
+			[
+				'label' => __( 'Image', 'elementor' ),
+				'type' => Controls_Manager::MEDIA,
+				'media_type' => 'image',
+				'should_include_svg_inline_option' => true,
+				'library_type' => 'image/svg+xml',
```

Comment:
The user can still choose a png file?

Reply:
Yes, there are 2 buttons:

![image](https://user-images.githubusercontent.com/32631382/109156202-3b9fab00-7779-11eb-8f45-c399aeb3955c.png)

The left one will accept PNGs